compiler/damlc/daml-compiler/src/DA/Daml/Compiler/DataDependencies.hs

```diff
@@ -468,18 +424,21 @@ errTooManyNameComponents cs =
     "Internal error: Dalf contains type constructors with more than two name components: " <>
     (T.unpack $ T.intercalate "." cs)
 
-convKind :: Env -> LF.Kind -> LHsKind GhcPs
+convKind :: Env -> LF.Kind -> Gen (LHsKind GhcPs)
 convKind env = \case
-    LF.KStar -> noLoc $ HsStarTy noExt False
-    LF.KNat -> noLoc $ mkGhcType "Nat"
-    LF.KArrow k1 k2 -> noLoc $ HsFunTy noExt (convKind env k1) (convKind env k2)
-
-convTyVarBinder :: Env -> (LF.TypeVarName, LF.Kind) -> LHsTyVarBndr GhcPs
+    LF.KStar -> pure . noLoc $ HsStarTy noExt False
+    LF.KNat -> noLoc <$> mkGhcType env "Nat"
```

Comment:
By the way, this is 100% a module reference we weren't tracking before, but since GHC.Types is pretty commonly depended on, it didn't cause problems in practice.

Reply:
Yeah, the new approach is a lot less fragile :+1: